pyramid_transactional_celery/transactional_task.py

```diff
@@ -85,19 +88,15 @@ def tpc_abort(self, transaction):
 
 
 class TransactionalTask(Task):
-    """A task whose execution is delayed until after the current transaction.
-    """
-    abstract = True
-
-    def original_apply_async(self, *args, **kwargs):
-        """Shortcut method to reach real implementation of
-        celery.Task.apply_sync"""
-        return super(TransactionalTask, self).apply_async(*args, **kwargs)
+    """A task whose execution is delayed until the current transaction gets committed."""
```

Comment:
Vai faltar apenas colocar o `call_from_tpc_finish` novamente nos retries. Você podia estender o método `retry` da task com um simples:
```suggestion
    """A task whose execution is delayed until the current transaction gets committed."""

    def retry(self, *args, **kwargs):
        super(TransactionalTask, self).retry(*args, call_from_tpc_finish=True, **kwargs)
```

Talvez esse param pudesse se chamar algo tipo `bypass_transaction_manager` :wink: